compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -53,7 +53,7 @@ trait QuotesAndSplices {
       ctx.error(em"Quotes require stable QuoteContext, but found non stable $qctx", qctx.sourcePos)
 
     val tree1 =
-      if ctx.mode.is(Mode.Pattern) && level == 0 then
+      if ctx.mode.is(Mode.Pattern) then
```

Comment:
Why remove `level == 0`?

Reply:
Because if we are in a pattern and we go in the other branch we might generate some non-pattern trees (after some error has been emitted) which leads to crashes later in typer.

Reply:
`tests/neg/i8052.scala` had this issue. With this change, the error message is also more informative.

Reply:
If `level == 0` is a correct check, it seems better to keep it, and fix crash & error message elsewhere. The reason is that we don't know whether the removal introduces crashes for more ill-formed programs or not.

Reply:
The point is that is vas not a correct check. The level checks happen later in the Staging phase.